monitor/guard.go

```diff
@@ -273,6 +290,39 @@ func (m *Monitor) guardTxHandler(
 	}
 }
 
+func (m *Monitor) guardJobHandler(description string, request *guard.Request) esTxManager.JobHandler {
+	guardState := guard.ChanStatus_Idle
+	if description == "IntendSettle" {
+		guardState = guard.ChanStatus_Settled
+	}
+	return func(jobID uuid.UUID, receipt *ethtypes.Receipt) error {
+		if receipt.Status == ethtypes.ReceiptStatusSuccessful {
+			log.Infof("%s transaction %x succeeded", description, receipt.TxHash)
+			guardProof := guard.NewGuardProof(
+				mainchain.Bytes2Cid(request.ChannelId),
+				mainchain.Hex2Addr(request.SimplexReceiver),
+				receipt.TxHash,
+				receipt.BlockNumber.Uint64(),
+				m.EthClient.Address,
+				guardState)
+			syncData := m.Transactor.CliCtx.Codec.MustMarshalBinaryBare(guardProof)
+			msg := sync.NewMsgSubmitChange(sync.GuardProof, syncData, m.EthClient.Client, m.Transactor.Key.GetAddress())
+			log.Infof("submit change tx: guard proof request %s", request)
+			m.Transactor.AddTxMsg(msg)
+			err := m.dbDelete(GetGuardKey(mainchain.Bytes2Cid(request.ChannelId), mainchain.Hex2Addr(request.SimplexReceiver)))
+			if err != nil {
+				log.Errorln("db Delete err", err)
+				return err
+			}
+		} else {
+			txErr := fmt.Errorf("%s transaction %x failed", description, receipt.TxHash)
+			log.Error(txErr)
+			return txErr
+		}
+		return nil
+	}
+}
```

Comment:
code duplicate with `guardTxHandler`, make the common code as a func

Reply:
Done.